core/app/models/spree/product.rb

```diff
@@ -36,6 +36,14 @@ class Product < Spree::Base
     has_many :product_promotion_rules, class_name: 'Spree::ProductPromotionRule'
     has_many :promotion_rules, through: :product_promotion_rules, class_name: 'Spree::PromotionRule'
 
+    has_many :promotions, through: :promotion_rules, class_name: 'Spree::Promotion'
```

Comment:
@Mafi88 I think not, as has_many will return records for each entry in through association which will be unique already.